optihood/energy_network.py

```diff
@@ -1218,15 +1288,17 @@ def createScenarioFile(configFilePath, excelFilePath, numberOfBuildings):
 
         _gsw.create_scenario_file(configFilePath, excelFilePath, numberOfBuildings)
 
-    def setFromExcel(self, filePath, numberOfBuildings, clusterSize={}, opt="costs", mergeLinkBuses=False, dispatchMode=False, includeCarbonBenefits=False):
+    def setFromExcel(self, filePath, numberOfBuildings, clusterSize={}, opt="costs", mergeLinkBuses=False, mergeBuses=None, mergeHeatSourceSink=False, dispatchMode=False, includeCarbonBenefits=False):
         # does Excel file exist?
         if not filePath or not os.path.isfile(filePath):
             logging.error("Excel data file {} not found.".format(filePath))
         logging.info("Defining the energy network from the excel file: {}".format(filePath))
         self._dispatchMode = dispatchMode
-        data = pd.ExcelFile(filePath)
         self._optimizationType = opt
-        nodesData = self.createNodesData(data, filePath, numberOfBuildings, clusterSize)
+        data = pd.ExcelFile(filePath)
+        initial_nodal_data = self.get_nodal_data_from_Excel(data)
+        # data.close()
```

Comment:
The tests shows that these files are left open if not explicitly closed.
Though I could not find any further usage of it in this method, closing the file at this point breaks the functionality.